src/server/req_manager.c

```diff
@@ -2876,6 +2878,41 @@ make_host_addresses_list(char *phost, u_long **pul)
 		return 0;
 }
 
+/**
+ * @brief
+ * 		remove the cached ip addresses of a mom from the host tree and the ipaddrs tree
+ *
+ * @param[in]	pmom - ptr to the mom info
+ *
+ * @return	error code
+ * @retval	0			- no error
+ * @retval	PBS error	- error
+ */
+int
+remove_mom_ipaddresses_list(mominfo_t *pmom)
+{
+	/* take ipaddrs from ipaddrs cache tree */
+	if (hostaddr_tree != NULL) {
+		struct pul_store *tpul;
+		u_long *pul;
+
+		if ((tpul = (struct pul_store *) find_tree(hostaddr_tree, pmom->mi_host)) != NULL) {
+			pul = tpul->pul;
+			while (*pul) {
+					tdelete2(*pul, pmom->mi_port, &ipaddrs);
+					pul++;
+			}
+
+			if (tree_add_del(hostaddr_tree, pmom->mi_host, NULL, TREE_OP_DEL) != 0)
+				return (PBSE_SYSTEM);
```

Comment:
Parentheses are not necessary here.

Reply:
removed

Reply:
I've thought about the use of parens with return after I saw many of them removed in a recent PR. I tend to favor their use in case you ever want to define a macro for return(). If you don't have the parens, then a macro won't work. As of now, there is no rule governing whether to use them or not other than following the local style as @subhasisb has done here.

Reply:
I agree that parens are useful for macros, but it makes more sense to me to put parens when defining them rather than when using.

Reply:
For this PR I am not going to insist on removing the parens so that it complies with the existing style in the file. I can open anther PR in the future and move the discussion there. Thanks @subhasisb and @mike0042.

Reply:
Looking at the current version of src/server/req_manager.c I don't see any consistency in the use of parens with return statements. Until there is a policy that mandates their use (or non-use), I don't see anything wrong with the change as it stands.

Reply:
I agree with @minghui-liu about the use of macros.  One of the perils of using macros is not using enough parenthese.  A macro of SQ(x) x * x is bad.  If you were to call SQ(x+1), you'll end up with x+1 * x+1 and end up with 2x+2.  SQ(x) (x) * (x) or even better ((x) * (x)).  I also agree with @mike0042 that until we have a standard for whether or not parenthese are needed for return statements, anything really goes.  Consistency with the file is the best we can hope for.